tests/flows/test_oidc-saml.py

```diff
@@ -98,3 +120,57 @@ def test_full_flow(self, satosa_config_dict, oidc_frontend_config, saml_backend_
+        user_id = "testuser1"
+
+        # proxy config
+        satosa_config_dict["FRONTEND_MODULES"] = [oidc_multi_value_frontend_config]
+        satosa_config_dict["BACKEND_MODULES"] = [saml_backend_config]
+        satosa_config_dict["INTERNAL_ATTRIBUTES"]["attributes"] = {attr_name: {"openid": [attr_name],
+                                                                               "saml": [attr_name]}
+                                                                   for attr_name in USERS[user_id]}
+        _, backend_metadata = create_entity_descriptors(SATOSAConfig(satosa_config_dict))
+
+        # application
+        test_client = Client(make_app(SATOSAConfig(satosa_config_dict)), BaseResponse)
+
+        # get frontend OP config info
+        provider_config = json.loads(test_client.get("/.well-known/openid-configuration").data.decode("utf-8"))
+
+        # create auth req
+        claims_request = ClaimsRequest(id_token=Claims(**{k: None for k in USERS[user_id]}))
+        req_args = {"scope": "openid", "response_type": "id_token", "client_id": CLIENT_ID,
+                    "redirect_uri": REDIRECT_URI, "nonce": "nonce",
+                    "claims": claims_request.to_json()}
+        auth_req = urlparse(provider_config["authorization_endpoint"]).path + "?" + urlencode(req_args)
+
+        # make auth req to proxy
+        proxied_auth_req = test_client.get(auth_req)
+        assert proxied_auth_req.status == "303 See Other"
+
+        # config test IdP
+        backend_metadata_str = str(backend_metadata[saml_backend_config["name"]][0])
+        idp_conf["metadata"]["inline"].append(backend_metadata_str)
+        fakeidp = FakeIdP(USERS, config=IdPConfig().load(idp_conf, metadata_construction=False))
+
+        # create auth resp
+        req_params = dict(parse_qsl(urlparse(proxied_auth_req.data.decode("utf-8")).query))
+        url, authn_resp = fakeidp.handle_auth_req(
+            req_params["SAMLRequest"],
+            req_params["RelayState"],
+            BINDING_HTTP_REDIRECT,
+            user_id,
+            response_binding=BINDING_HTTP_REDIRECT)
+
+        # make auth resp to proxy
+        authn_resp_req = urlparse(url).path + "?" + urlencode(authn_resp)
+        authn_resp = test_client.get("/" + authn_resp_req)
+        assert authn_resp.status == "303 See Other"
+
+        # verify auth resp from proxy
+        resp_dict = dict(parse_qsl(urlparse(authn_resp.data.decode("utf-8")).fragment))
+        signing_key = RSAKey(key=rsa_load(oidc_multi_value_frontend_config["config"]["signing_key_path"]),
+                             use="sig", alg="RS256")
+        id_token_claims = JWS().verify_compact(resp_dict["id_token"], keys=[signing_key])
+        assert all((k, v) in {ck:cv if isinstance(cv,list) else [cv] for ck, cv in id_token_claims.items()}.items() for k, v in USERS[user_id].items())
```

Comment:
This is a bit complex. I'm adding whitespace to make it readable

```python
assert all(
  (k, v) in {
    ck: cv if isinstance(cv,list) else [cv]
    for ck, cv in id_token_claims.items()
  }.items()
  for k, v in USERS[user_id].items())
```

What you're trying to do is figuring out if _all items_ from `USERS` match an item in `id_token_claims` after a transformation. This means that `USERS` is a subset of (the transformed) `id_token_claims`.

```python
claims_dict =  {
    ck: cv if isinstance(cv,list) else [cv]
    for ck, cv in id_token_claims.items()
}
claims_set = set(claims_dict.items())
users_set = set(USERS[user_id].items())
assert users_set.issubset(claims_set)
```

I think, though, that what you're really after is equality of the dicts.

```python
claims_dict =  {
    ck: cv if isinstance(cv,list) else [cv]
    for ck, cv in id_token_claims.items()
}
assert claims_dict == USERS[user_id]
```